jumpscale/packages/vdc_dashboard/bottle/controller.py

```diff
@@ -0,0 +1,275 @@
+
+    # Payment
+    success, _, _ = vdc.show_external_node_payment(bot=None, size=node_flavor, public_ip=False)
+    if not success:
+        abort(400, "Not enough funds in prepaid wallet to add node")
+
+    old_wallet = deployer._set_wallet(vdc.prepaid_wallet.instance_name)
+    try:
+        wids = deployer.add_k8s_nodes(node_flavor, public_ip=False)
+        deployer._set_wallet(old_wallet)
+        return HTTPResponse(
+            j.data.serializers.json.dumps(wids), status=201, headers={"Content-Type": "application/json"}
+        )
+    except Exception as e:
+        abort(400, f"failed to add nodes to your cluster. due to error {str(e)}")
+
+
+@app.route("/api/controller/node/delete", method="POST")
+@controller_authorized()
+def delete_node():
+    """
+    request body:
+        password
+        wid
+
+    Returns:
+        status
+    """
+    data = j.data.serializers.json.loads(request.body.read())
+    vdc_password = data.get("password")
+    wid = data.get("wid")
+    if not all([wid]):
+        abort(400, "Error: Not all required params were passed.")
+
+    vdc = get_vdc()
+    vdc.load_info()
+    deployer = vdc.get_deployer(password=vdc_password)
+
+    try:
+        deployer.delete_k8s_node(wid)
+    except j.exceptions.Input:
+        abort(400, "Error: Failed to delete workload")
+
+    return HTTPResponse(
+        j.data.serializers.json.dumps({"result": True}), status=200, headers={"Content-Type": "application/json"}
+    )
+
+
+@app.route("/api/controller/zdb/list", method="POST")
+@controller_authorized()
+def list_zdbs():
+    """
+    request body:
+        password
+
+    Returns:
+        zdbs: string
+    """
+    data = j.data.serializers.json.loads(request.body.read())
+    vdc_dict = _get_vdc_dict()
+
+    return HTTPResponse(
+        j.data.serializers.json.dumps(vdc_dict["s3"]["zdbs"]), status=200, headers={"Content-Type": "application/json"}
+    )
+
+
+@app.route("/api/controller/zdb/add", method="POST")
+@controller_authorized()
+def add_zdb():
+    """
+    request body:
+        password
+        capacity
+        farm(optional)
+
+    Returns:
+        wids: list of wids
+    """
+    data = j.data.serializers.json.loads(request.body.read())
+    capacity = data.get("capacity")
+    farm = data.get("farm")
+
+    if not all([capacity]):
+        abort(400, "Error: Not all required params were passed.")
+
+    vdc = get_vdc()
+    vdc.load_info()
+
+    zdb_monitor = vdc.get_zdb_monitor()
+    if not farm:
+        zdb_farms = zdb_monitor.get_zdb_farm_names()
+        farm = random.choice(zdb_farms)
+    try:
+        wids = zdb_monitor.extend(required_capacity=capacity, farm_names=[farm], wallet_name="prepaid_wallet")
+        return HTTPResponse(
+            j.data.serializers.json.dumps(wids), status=201, headers={"Content-Type": "application/json"}
+        )
+    except Exception as e:
+        abort(500, f"Error: Failed to deploy zdb due to the following {str(e)}")
```

Comment:
I think we can throw 500 but not with the exception details, alternatively we can log error it, to encapsulate the errors for our side